src/color-map/ColorMapFilter.ts

```diff
@@ -54,13 +55,36 @@ export class ColorMapFilter extends Filter
     private _scaleMode: SCALE_MODE = 'linear';
     private _colorMap!: ColorMapTexture;
 
-    constructor(options: ColorMapFilterOptions)
+    constructor(options: ColorMapFilterOptions);
+    /**
+     * @deprecated since 8.0.0
+     *
+     * @param {HTMLImageElement|HTMLCanvasElement|PIXI.BaseTexture|PIXI.Texture} [colorMap] - The
+     *        colorMap texture of the filter.
+     * @param {boolean} [nearest=false] - Whether use NEAREST for colorMap texture.
+     * @param {number} [mix=1] - The mix from 0 to 1, where 0 is the original image and 1 is the color mapped image.
+     */
+    constructor(colorMap: ColorMapTexture, nearest?: boolean, mix?: number);
+    constructor(...args: [ColorMapFilterOptions] | [ColorMapTexture, boolean?, number?])
     {
+        let options = args[0] ?? {};
+
+        if (options instanceof Texture || options instanceof TextureSource)
+        {
+            // eslint-disable-next-line max-len
+            deprecation(v8_0_0, 'ColorMapFilter constructor params are now options object. See params: { colorMap, nearest, mix }');
+
+            options = { colorMap: options };
+
+            if (args[1]) options.nearest = args[1];
+            if (args[2]) options.mix = args[2];
```

Comment:
More important here for args[2] which can be 0 and would not get set.
```suggestion
            if (args[1] !== undefined) options.nearest = args[1];
            if (args[2] !== undefined) options.mix = args[2];
```

Reply:
Ooooo, amazing catch - thank you!